src/main/java/com/redhat/devtools/intellij/tektoncd/actions/pipeline/CreatePipelineAction.java

```diff
@@ -13,26 +13,50 @@
 import com.google.common.base.Strings;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.redhat.devtools.intellij.tektoncd.actions.TektonAction;
+import com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService;
 import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
 import com.redhat.devtools.intellij.tektoncd.tree.PipelinesNode;
 import com.redhat.devtools.intellij.tektoncd.utils.VirtualFileHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.swing.tree.TreePath;
 
+import java.io.IOException;
+
 import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_PIPELINES;
+import static com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.ActionMessage;
+import static com.redhat.devtools.intellij.telemetry.core.util.AnonymizeUtils.anonymizeResource;
 
 
 public class CreatePipelineAction extends TektonAction {
-    public CreatePipelineAction() { super(PipelinesNode.class); }
+
+    private static final Logger logger = LoggerFactory.getLogger(CreatePipelineAction.class);
+
+    public CreatePipelineAction() {
+        super(PipelinesNode.class);
+    }
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
         PipelinesNode item = getElement(selected);
         String namespace = item.getParent().getName();
         String content = getSnippet("Tekton: Pipeline");
 
-        if (!Strings.isNullOrEmpty(content)) {
-            VirtualFileHelper.createAndOpenVirtualFile(anActionEvent.getProject(), namespace,namespace + "-newpipeline.yaml", content, KIND_PIPELINES, item);
+        ActionMessage telemetry = TelemetryService.instance()
+                .action("create pipeline");
+        if (Strings.isNullOrEmpty(content)) {
+            telemetry.error("snippet content empty").send();
+        } else {
+            telemetry.send();
```

Comment:
Should this be removed?

Reply:
if missing/empty snippets if nothing we'd want to report, yes. I was inserting this because I thought that we'd report that the action failed. Thoughts?